ws-client/src/transport.rs

```diff
@@ -46,6 +46,17 @@ impl Host {
 	}
 }
 
+/// String representation of the path of the URL.
+#[derive(Clone, Debug)]
+pub struct Path(String);
+
+impl Path {
+	/// Extracts a string slice from the inner String.
+	pub fn as_str(&self) -> &str {
+		self.0.as_str()
+	}
+}
```

Comment:
Not sure if this type is pulling its weight, we could probably do without it yes?

Reply:
let's do `PeerInfo` or `ParsedWsAddress` and group all in type instead.

I just don't like have a bunch of Strings in tuple, it's error-prone.

Reply:
Maybe "peer" is a poor choice, this is not really a p2p context. How about "Remote": `RemoteInfo`/`RemoteAddress`?

Reply:
RemoteWsTarget ok?